atdgen/src/validate.ml

```diff
@@ -24,10 +24,13 @@ type validate_repr = (string option * bool)
 
 let make_full_validator s =
   sprintf "\
-    fun path x -> \
-      if ( %s ) x then None \
-      else Some (Atdgen_runtime.Util.Validation.error path)"
-    s
+  fun path x ->
+    let msg = \"Failed check by %s\" in
+    if (%s) x then
+      None
+    else
+      Some (Atdgen_runtime.Util.Validation.error ~msg path)"
+    s s
```

Comment:
The first `%s` is in a string literal, so I think it should quoted. Check what you're getting if the argument of `make_full_validator` contains double quotes, like `"(fun s -> s = \"abc\")"`.

Reply:
Lokks good to me other than that.

Reply:
That's good thinking. I'll add some test cases for that.